### PR TITLE
SBP 2.0

### PR DESCRIPTION
**Pull request by mfine**
New conversions for SBP 2.0.

/cc @ljbade @mookerji

### CODE REVIEW

File: haskell/src/Data/RTCM3/SBP.hs [hunk 1]
```diff
@@ -296,126 +296,99 @@ toCn0_L1 = (^. gpsL1ExtObservation_cnr)
...
-  return PackedObsContentDepC
-    { _packedObsContentDepC_P    = toP_L1 l1 l1e
-    , _packedObsContentDepC_L    = toL_L1 l1 l1e
-    , _packedObsContentDepC_cn0  = toCn0_L1 l1e
-    , _packedObsContentDepC_lock = lock
-    , _packedObsContentDepC_sid  = sid
+  return PackedObsContent
+    { _packedObsContent_P     = toP_L1 l1 l1e
+    , _packedObsContent_L     = toL_L1 l1 l1e
+    , _packedObsContent_D     = errorWithStackTrace "D"
```
Comment (mfine) [inline haskell/src/Data/RTCM3/SBP.hs]: *TODO* need to calculate Doppler from RTCM.

/cc @denniszollo

File: haskell/src/Data/RTCM3/SBP.hs [hunk 2]
```diff
@@ -296,126 +296,99 @@ toCn0_L1 = (^. gpsL1ExtObservation_cnr)
...
+    , _packedObsContent_cn0   = toCn0_L1 l1e
+    , _packedObsContent_lock  = l1 ^. gpsL1Observation_lockTime
+    , _packedObsContent_sid   = sid
+    , _packedObsContent_flags = errorWithStackTrace "flags"
```
Comment (mfine) [inline haskell/src/Data/RTCM3/SBP.hs]: Which flags should we use here?

File: haskell/src/Data/RTCM3/SBP.hs (hunk 1 above)
Comment (ljbade) [inline haskell/src/Data/RTCM3/SBP.hs]: I believe RTCM doesn't have doppler? if so fill with zeros and don't set doppler valid flag

File: haskell/src/Data/RTCM3/SBP.hs [hunk 3]
```diff
@@ -296,126 +296,99 @@ toCn0_L1 = (^. gpsL1ExtObservation_cnr)
...
-    , _packedObsContentDepC_L    = toL_L1 l1 l1e
-    , _packedObsContentDepC_cn0  = toCn0_L1 l1e
-    , _packedObsContentDepC_lock = lock
-    , _packedObsContentDepC_sid  = sid
+  return PackedObsContent
+    { _packedObsContent_P     = toP_L1 l1 l1e
+    , _packedObsContent_L     = toL_L1 l1 l1e
+    , _packedObsContent_D     = errorWithStackTrace "D"
+    , _packedObsContent_cn0   = toCn0_L1 l1e
+    , _packedObsContent_lock  = l1 ^. gpsL1Observation_lockTime
```
Comment (ljbade) [inline haskell/src/Data/RTCM3/SBP.hs]: Yeah I think you will need to translate here, we are using the MSM version of lock time which I think might use slightly different encoding to the older RTCM 3 messages

**Comment by mfine** [comment]
Updated with lock time mapping.

**Comment by mfine** [comment]
This should be complete now.

**Comment by ljbade** [comment]
@mfine should we also bump sbp to latest 2.1.1 version?

**Pull request merged**